src/Playwright/Core/Browser.cs

```diff
@@ -140,12 +141,14 @@ public async Task<IPage> NewPageAsync(BrowserNewPageOptions options = default)
                 Proxy = options.Proxy,
                 StorageState = options.StorageState,
                 StorageStatePath = options.StorageStatePath,
+                BaseURL = options.BaseURL,
             };
 
             var context = (BrowserContext)await NewContextAsync(contextOptions).ConfigureAwait(false);
 
             var page = (Page)await context.NewPageAsync().ConfigureAwait(false);
             page.OwnedContext = context;
+            context.BaseUrl = options.BaseURL;
```

Comment:
don't you store the context options already on the context when creating a new context? Thats how its done upstream and in Python.

Reply:
No, we don't store the `options` classes AFAICS.

Reply:
This is a nit, but I'm with Max - we could store the entire thing for simplicity.